packages/pigment-css-react/README.md

```diff
@@ -854,3 +854,216 @@ function App() {
+
+Assemble the component using the slots created in the previous step:
+
+```js
+// /path/to/Stat.js
+import * as React from 'react';
+
+const StatRoot = styled('div', {
+  name: 'MuiStat',
+  slot: 'root',
+})(…);
+
+const StatValue = styled('div', {
+  name: 'MuiStat',
+  slot: 'value',
+})(…);
+
+const StatUnit = styled('div', {
+  name: 'MuiStat',
+  slot: 'unit',
+})(…);
+
+const Stat = React.forwardRef(function Stat(props, ref) {
+  const { value, unit, ...other } = props;
+
+  return (
+    <StatRoot ref={ref} {...other}>
+      <StatValue>{value}</StatValue>
+      <StatUnit>{unit}</StatUnit>
+    </StatRoot>
+  );
+});
+
+export default Stat;
+```
+
+### 3. Style the slot with ownerState
+
+When you need to style the slot-based props or internal state, wrap them in the `ownerState` object and pass it to each slot as a prop.
+The `ownerState` is a special name that will not spread to the DOM via the `styled` API.
+
+Add a `variant` prop to the `Stat` component and use it to style the `root` slot, as shown below:
+
+```diff
+  const Stat = React.forwardRef(function Stat(props, ref) {
++   const { value, unit, variant, ...other } = props;
++
++   const ownerState = { ...props, variant };
+
+    return (
+-      <StatRoot ref={ref} {...other}>
+-        <StatValue>{value}</StatValue>
+-        <StatUnit>{unit}</StatUnit>
+-      </StatRoot>
++      <StatRoot ref={ref} ownerState={ownerState} {...other}>
++        <StatValue ownerState={ownerState}>{value}</StatValue>
++        <StatUnit ownerState={ownerState}>{unit}</StatUnit>
++      </StatRoot>
+    );
+  });
+```
+
+Then you can read `ownerState` in the slot to style it based on the `variant` prop.
+
+```diff
+  const StatRoot = styled('div', {
+    name: 'MuiStat',
+    slot: 'root',
+-  })(({ theme }) => ({
++  })(({ theme, ownerState }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '1rem',
+    padding: '0.75rem 1rem',
+    backgroundColor: theme.colors.primary.background,
+    borderRadius: theme.radius.xs,
+    boxShadow: theme.shadow.sm,
+    letterSpacing: '-0.025em',
+    fontWeight: 600,
++   variants: [
++     {
++       props: { variant: 'outlined' },
++       style: {
++         border: `2px solid #e9e9e9`,
++       },
++     },
++   ],
+  }));
+```
+
+That's it! You've successfully created a statistics component with Pigment CSS. The next step is to upload your component to a package registry to let other people use it.
+
+### Consumer usage
+
+The developers who want to use your component need to install your package and Pigment packages based on their [framework](#start-with-nextjs).
+
+```bash
+npm install your-package-name @pigment-css/react
+npm install -D @pigment-css/nextjs-plugin
+```
+
+Then, they need to set up Pigment CSS in their project and import the stylesheet in the root layout file:
+
+```js
+// framework config file, for example next.config.js
+const { withPigment } = require('@pigment-css/nextjs-plugin');
+
+const theme = {
```

Comment:
Maybe we can also emphasize that lub authors can also export their pre configured theme that can be directly imported and passed to the config.

Reply:
I decided not to mention the theme configuration at the beginning. I think the theme should be added after teaching how to create a component.